### PR TITLE
Support Jenkins Pipeline

### PR DESCRIPTION
**Pull request by Dahlgren**
```
pipeline {
  agent any

  post {
    always {
      discordSend description: 'Jenkins Pipeline Build', footer: 'Footer Text', link: BUILD_URL, successful: currentBuild.resultIsBetterOrEqualTo('SUCCESS'), title: JOB_NAME, webhookURL: 'Webhook URL'
    }
  }
}

```

### CODE REVIEW

**Comment by jupjohn** [comment]
Tell me why exactly you've made every field of the webhook editable? I'm confused. It also doesn't support the current features like disabling linking, enabling artifact lists etc...

**Comment by Dahlgren** [comment]
Pipeline steps are different from the usual Notifier plugins that are accessible post build. The `discordSend` can be invoked anytime during the build and multiple times.

Only webhookURL must be set, the rest is optional and can be omitted.

By allowing the Pipeline script to define the texts sent to Discord it's possible to send different messages for different stages in the pipeline. We use this to send notifications for each failing parallel build in our matrix pipeline. Once all parallel builds are completed we can also send a final notification for the entire pipeline in the post steps such as in the example in the dscription.

Some reasons for not using the other perform method,
- The build result might not be resolved yet
- AbstractBuild is not fully compatible with the newer base Run type. Things such as `getChangeSet()` might not be available, it's only present for some subclasses.
- Would be restricted to run once for the end of the pipeline

By having this more open allows for more flexibility in pipeline usage. Most notification plugins such as slack works the same, see the example in it's repo https://github.com/jenkinsci/slack-plugin

I'm open for making changes in the PR, it doesn't have to be this way :)

File: src/main/java/nz/co/jammehcow/jenkinsdiscord/DiscordPipelineStep.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        @Override
+        public String getFunctionName() {
+            return "discordSend";
+        }
+
+        @Nonnull
+        @Override
+        public String getDisplayName() {
+            return "Send message to Discord Webhook";
```
Comment (jupjohn) [inline src/main/java/nz/co/jammehcow/jenkinsdiscord/DiscordPipelineStep.java]: Would prefer "Send an embed message to Webhook URL"

File: src/main/resources/nz/co/jammehcow/jenkinsdiscord/DiscordPipelineStep/help-description.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<div>
+    Main message text
```
Comment (jupjohn) [inline src/main/resources/nz/co/jammehcow/jenkinsdiscord/DiscordPipelineStep/help-description.html]: Too brief, needs to explain a bit more.

File: src/main/java/nz/co/jammehcow/jenkinsdiscord/DiscordPipelineStep.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+public class DiscordPipelineStep extends AbstractStepImpl {
+
```
Comment (jupjohn) [inline src/main/java/nz/co/jammehcow/jenkinsdiscord/DiscordPipelineStep.java]: It's nothing major, but removing that newline under the class' curly brace would be nice. Fits with the rest of the classes.

**Review by jupjohn — CHANGES_REQUESTED** [review]

**Comment by jupjohn** [comment]
Apart from those little things, if you could add a section in the README that explains how to use it that'd be great. I'll merge afterwards.

File: src/main/resources/nz/co/jammehcow/jenkinsdiscord/DiscordPipelineStep/help-description.html (hunk 2 above)
Comment (Dahlgren) [reply src/main/resources/nz/co/jammehcow/jenkinsdiscord/DiscordPipelineStep/help-description.html]: I'd add some notes about markdown at least

**Comment by Dahlgren** [comment]
I'll add some README documentation about how to use in Pipeline builds

**Comment by Dahlgren** [comment]
Please check if you want any further changes 😃

**Comment by jupjohn** [comment]
Thanks for the addition.

**Pull request merged**